### PR TITLE
Make SourceKit-LSP’s documentation more discoverable

### PR DESCRIPTION
**Pull request by ahoppen**

### CODE REVIEW

File: README.md [hunk 1]
```diff
@@ -6,13 +6,15 @@ SourceKit-LSP is an implementation of the [Language Server Protocol](https://mic
 
 https://www.swift.org/tools has a list of popular editors that support LSP and can thus be hooked up to SourceKit-LSP to provide intelligent editor functionality as well as set-up guides.
 
+> [!IMPORTANT]
+> SourceKit-LSP does not update its global index in the background or build Swift modules in the background. Thus, a lot of cross-module or global functionality is limited if the project hasn't been built recently. To update the index or rebuild the Swift modules, build the project or enable the experimental background indexing as described in [Enable Experimental Background Indexing](Documentation/Enable%20Experimental%20Background%20Indexing.md).
```
Comment (MaxDesiatov) [inline README.md:10]: IMO "your project" makes it clearer that you have to rebuild your own project, not SourceKit-LSP from sources itself.
```suggestion
> SourceKit-LSP does not update its global index in the background or build Swift modules in the background. Thus, a lot of cross-module or global functionality is limited if the project hasn't been built recently. To update the index or rebuild the Swift modules, build your project or enable the experimental background indexing as described in [Enable Experimental Background Indexing](Documentation/Enable%20Experimental%20Background%20Indexing.md).
```

**Comment by ahoppen** [comment]
@swift-ci Please test

File: Documentation/README.md [hunk 2]
```diff
@@ -0,0 +1,22 @@
...
+
+- [Configuration File](Configuration%20File.md)
+- [Diagnose Bundle](Diagnose%20Bundle.md)
+- [Editor Integration](Editor%20Integration.md)
+- [Enable Experimental Background Indexing](Enable%20Experimental%20Background%20Indexing.md)
+- [Environment Variables](Environment%20Variables.md)
+
+## Developer Documentation
+
+The following documentation documents are primarily intended for developers of SourceKit-LSP.
```
Comment (bnbarham) [inline Documentation/README.md:15]: Should we link to these from CONTRIBUTING.md instead?

**Review by bnbarham — APPROVED** [review]

File: Documentation/README.md (hunk 2 above)
Comment (ahoppen) [reply Documentation/README.md]: I split the documentation into the `Documentation` folder (documentation for users of SourceKit-LSP) and `Contributor Documentation` (for contributors) and link to `Contributor Documentation` from CONTRIBUTING.md.

**Comment by ahoppen** [comment]
@swift-ci Please test

**Comment by ahoppen** [comment]
@swift-ci Please test Windows

**Review by MaxDesiatov — APPROVED** [review]

**Pull request merged**